### PR TITLE
[tcgc] add UsageFlags.JsonMergePatch

### PR DESCRIPTION
**Pull request by iscai-msft**

### CODE REVIEW

**Comment by tadelesh** [comment]
I'd like to hold this PR before we could meet all requirements from Java and .NET from this [issue](https://github.com/Azure/typespec-azure/issues/460).

**Comment by iscai-msft** [comment]
@tadelesh confirmed with the rest of the dpg crew that we want a `JsonMergePatch` usage. A good amount of languages generate extra info for the model in that case. We will still mark it as `Input`, so they're is no rule that both can't be set

File: packages/typespec-client-generator-core/src/types.ts [hunk 1]
```diff
@@ -1264,14 +1264,18 @@ function updateTypesFromOperation(
...
-      checkAndGetClientType(context, httpOperation.parameters.body.type, operation)
-    );
+  const httpBody = httpOperation.parameters.body;
+  if (httpBody) {
+    const bodies = diagnostics.pipe(checkAndGetClientType(context, httpBody.type, operation));
     if (generateConvenient) {
       bodies.forEach((body) => {
         updateUsageOfModel(context, UsageFlags.Input, body);
       });
+      if (httpBody.contentTypes.includes("application/json-merge-patch")) {
```
Comment (srnagar) [inline packages/typespec-client-generator-core/src/types.ts:1274]: The content type should be `application/merge-patch+json` - https://datatracker.ietf.org/doc/html/rfc7386

```suggestion
      if (httpBody.contentTypes.includes("application/merge-patch+json")) {
```

**Review by srnagar — APPROVED** [review]

File: packages/typespec-client-generator-core/src/interfaces.ts [hunk 2]
```diff
@@ -529,4 +529,5 @@ export enum UsageFlags {
   Input = 1 << 1,
   Output = 1 << 2,
   Versioning = 1 << 3,
```
[the remark this answers is not in the document]
Comment (iscai-msft) [reply packages/typespec-client-generator-core/src/interfaces.ts:531]: yep, that's a separate pr, let me send!

File: packages/typespec-client-generator-core/src/types.ts (hunk 1 above)
Comment (iscai-msft) [reply packages/typespec-client-generator-core/src/types.ts:1274]: oh oops, thank you so much for hte catch!

**Review by joheredi — APPROVED** [review]

**Pull request merged**